test/models/user_test.exs

```diff
@@ -69,12 +69,37 @@ defmodule Gannbaruzoi.UserTest do
     end
   end
 
+  describe "valid_token?" do
+    test "returns true or false" do
+      email = "hey-man@gmail.com"
+      insert!(:user, email: email)
+      auth = Repo.get_by!(User, email: email)
+             |> User.build_session()
+             |> Repo.update!
+             |> Map.get(:auth)
+
+      user = Repo.get_by!(User, email: email)
+      now = DateTime.utc_now() |> DateTime.to_unix()
+      fifteen_days_later = now + 60 * 60 * 24 * 14
+
+      assert User.valid_token?(user, auth.client, auth.token)
+      refute User.valid_token?(user, "wrong client", auth.token)
+      refute User.valid_token?(user, auth.client, "worng token")
+      refute User.valid_token?(user, auth.client, auth.token,
+                               fifteen_days_later)
+    end
+  end
+
   describe "match_password?/2" do
-    test "deletes client" do
+    test "returns true or false" do
       password = "alice1234"
-      user = insert_user(email: "hey@gmail.com", password: password)
+      email = "hey!@gmail.com"
+      %User{}
+      |> User.changeset(%{email: email, password: password})
```

Comment:
I can not use factory here. if password is not in changeset ecto just ignore it.